platform/darwin/src/NSDictionary+MGLAdditions.mm

```diff
@@ -0,0 +1,47 @@
+#import "NSDictionary+MGLAdditions.h"
+
+#import "NSExpression+MGLAdditions.mm"
+
+@implementation NSDictionary (MGLAdditions)
+
+- (mbgl::PropertyMap)mbgl_propertyMap {
+    return [self propertyMapWithDictionary:self];
+}
+
+- (mbgl::PropertyMap)propertyMapWithDictionary:(NSDictionary *)dictionary {
```

Comment:
Remove the `dictionary` parameter and use `self` instead. Below, call the method recursively on `dictionary[key]` rather than on `self`. Then rename this method to `-mgl_propertyMap` (not `-mbgl_propertyMap` – MGL is the class prefix for this target).